assets/dev/js/editor/commands/internal/load.js

```diff
@@ -42,7 +42,15 @@ export class Load extends CommandInternalBase {
 		}
 
 		if ( elementor.loaded ) {
-			$e.internal( 'editor/documents/attach-preview' );
+			// TODO: Find better solution - Fix issue when globals does not render after saving from kit.
+			// The issue is that the css-parser is depends upon cache and cache is not available during this time.
+			if ( ! $e.data.getCache( $e.components.get( 'globals' ), 'globals/index' ) ) {
+				$e.data.get( 'globals/index' ).then( () =>
+					$e.internal( 'editor/documents/attach-preview' )
+				);
+			} else {
+				$e.internal( 'editor/documents/attach-preview' );
+			}
```

Comment:
Instead of all this logic, since `$e.data.get( 'globals/index' )` retrieves the cache if it exists, you only need to always attach the preview after getting of `globals/index`:

```suggestion
			$e.data.get( 'globals/index' ).then( () =>
				$e.internal( 'editor/documents/attach-preview' )
			);
```